src/utils/Client.ts

```diff
@@ -71,31 +74,67 @@ export class Client extends Discord.Client {
 
 	public async addSlashCommands(path: string): Promise<this> {
 		try {
+			// eslint-disable-next-line no-await-in-loop
+			while (!this.user) await this.wait(500);

```

Comment:
This still makes me laugh every time I see it

Reply:
Could you explain better? Actually, I think that if I just do a `while (!this.user) {}` it will just get stuck inside of the `while` loop as no functions run == the while loop does not end.

Reply:
I'm not sure but I think it's the same. It executes a function continuosly and the only thing that changes is what's inside it.

Reply:
I may actually test it if I don't run in some weird race conditions through my tests...

Reply:
Yes, it actually falls in an infinite loop. I used this dummy code, but the result should be the same with that too.

```js
let a = "";

setTimeout(() => {
  a = "hello";
}, 500);

while (!a) {}

console.log(a);
```

Reply:
Apart from the fact that your implementation is totally wrong XD, yeah I think it's useless the same and the best solution would be to create a promise that resolves when the client is ready

Reply:
Now that I think of it, it's not so difficult. You should create two properties in the client:
```ts
// Assume we're in the class
private resolvePromise: (arg: unknown) => void;

readyPromise = new Promise(resolve => (this.resolvePromise = resolve)); // Deal with Eslint
```
And then in the constructor or wherever...
```ts
this.once("ready", this.resolvePromise);
```
So that instead of that while loop you could just `await this.readyPromise`! Not sure if you're ok to implement it tho xD

Reply:
I could do it, but that requires some little more work. Will work on it though.

Reply:
> Will work on it though.
Good joke ahahah

Reply:
Okay, I will probably work on this.